python-frontend/src/main/java/org/sonar/plugins/python/api/tree/Name.java

```diff
@@ -22,7 +22,7 @@
 /**
  * See https://docs.python.org/3/reference/expressions.html#atom-identifiers
  */
-public interface Name extends Expression, HasSymbol {
+public interface Name extends Expression, HasSymbol, Pattern {
```

Comment:
I'm not sure that's a good change. That would mean that:
* there's no easy way to visit all capture patterns.
* we have something which implements `Pattern` everywhere we have an identifier (I know that's what we already do with `Name` implementing `Expression`). I don't know if it's a real problem.

An alternative would be to introduce a `CapturePattern` class which would wrap an instance of `Name`. 
Maybe we can discuss that.

Reply:
Following our discussion, I will create `CapturePattern` wrapper